tools/fix_timepoint_date_problems.php

```diff
@@ -1,531 +1,673 @@
-#!/usr/bin/env php
-<?php
-#!/usr/bin/php
-/**
- * @version $Id: fix_timepoint_date_problems.php,v 1.21 2006/05/05 15:42:25 dario Exp $
- *
- * This is the tool to diagnose and correct the date problems in a candidate profile and add missing instruments to the bvl battery.
- *
- * Currently the tool does the following:
- * - corrects DOB and EDC (candidate table)
- * - corrects Date_screening, Date_visit (session table)
- * - corrects bvl batteries by:
- *    adding records in the flag table
- *    adding records in the instrument tables
- *    adding records into the feedback_bvl* tables (creates new feedback)
- *    writes a log entry
- *    sending automatic email notification - to be completed later
- *
- * The procedure to correct the problems is:
- * run w/ fix_date option to UPDATE the date fields in the candidate/session table (as needed)
- * run w/ diagnose option to get the difference in batteries
- * run the tool w/ add_instrument option to add missing instruments, and create new feedback
- * send email to PSC
- * close Mantis bug
- *
- * ways to use this script:
- * -- get a help screen
- * fix_timepoint_date_problems.php help
- *
- * -- to fix dates (when fixing stage dates, sessionID must be specified)
- * fix_timepoint_date_problems.php fix_date <candID> <newCorrectDate> <dob/edc>
- * fix_timepoint_date_problems.php fix_date <candID> <newCorrectDate> <screening/visit> <sessionID>
- *
- * -- when diagnosing a problem (if only CandID is specified, all timepoints are diagnosed
- * fix_timepoint_date_problems.php diagnose <candID> [<sessionID>]
- * fix_timepoint_date_problems.php diagnose <candID> [<newCorrectDate> <dob/edc/>]
- * fix_timepoint_date_problems.php diagnose <candID> [<sessionID> <newCorrectDate> <screening/visit>]
- *
- * -- to fix bvl battery (run only once the dates are fixed)
- * fix_timepoint_date_problems.php add_instrument  <candID> <sessionID> <test_name>
- *
- * @package behavioural
- */
-use LORIS\StudyEntities\Candidate\CandID;
-
-set_include_path(get_include_path().":../project/libraries:../php/libraries:");
-
-// path to config file
-$configFile = "../project/config.xml";
-
-require_once __DIR__ . "/../vendor/autoload.php";
-$client = new NDB_Client();
-$client->makeCommandLine();
-$client->initialize($configFile);
-
-$db = \Database::singleton();
-
-/**
- * HELP SCREEN
- * display and stop processing if action=help
- */
-if (empty($argv[1]) || $argv[1] == 'help') {
-    fwrite(STDERR, "Usage: \n\n");

```

Comment:
This is changing the destination of the error message. It should still be going to stderr if the usage was incorrect.